backend/app/services/knowledge/orchestrator.py

```
@@ -1467,6 +1467,348 @@ def update_document_content(
+        """
+        if source_type == "text":
+            if not content:
+                raise ValueError("content is required for source_type='text'")
+            return content.encode("utf-8"), _normalize_file_extension(file_extension)
+
+        elif source_type == "file":
+            if not file_base64:
+                raise ValueError("file_base64 is required for source_type='file'")
+            try:
+                binary_data = base64.b64decode(file_base64)
+            except Exception as e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)